dynamics/applications/NextsimMevp.cpp

```diff
@@ -0,0 +1,357 @@
+
+    Nextsim::ParametricGrid gridIn;
+    Nextsim::ParaGridIO* readIO = new Nextsim::ParaGridIO(gridIn);
+    gridIn.setIO(readIO);
+
+    Module::Module<Nextsim::IStructure>::setImplementation("ParametricGrid");
+
+    Nextsim::ModelState state = gridIn.getModelState("nextsimMEVPstart.nc");
+    size_t NX = Nextsim::ModelArray::definedDimensions.at(Nextsim::ModelArray::Dimension::X).length;
+    // Let's ASSUME it's square
+    //! Define the spatial mesh
+    Nextsim::ModelArray& coords = state.data.at(Nextsim::coordsName);
+    Nextsim::ParametricMesh smesh(NX, NX, state.data.at(Nextsim::coordsName).data().matrix());
+
+    //! Define the spatial mesh
+    Nextsim::ParametricMesh smeshOG;
+    smeshOG.readmesh(meshfile);
+
+    // Assert some facts about the vertex positions
+    assert(smesh.nx == NX);
+    assert(smesh.ny == NX);
+    assert(smesh.nelements == NX * NX);
+    assert(smesh.nnodes == (NX + 1) * (NX + 1));
+    assert(smesh.vertices(0, 0) == smesh.vertices(1, 0));
+    assert(smesh.vertices(0, 1) != smesh.vertices(1, 1));
+    assert(smesh.vertices(0, 0) != smesh.vertices(NX+1, 0));
+    assert(smesh.vertices(0, 1) == smesh.vertices(NX+1, 1));
+
+    //! Compose name of output directory and create it
+    std::string resultsdir = "BenchmarkNS_" + std::to_string(CG) + "_" + std::to_string(DGadvection) + "_" + std::to_string(DGstress)
+        + "__" + std::to_string(NX);
+    std::filesystem::create_directory(resultsdir);
+
+    //! Main class to handle the momentum equation. This class also stores the CG velocity vector
+    Nextsim::CGParametricMomentum<CG, DGstress> momentum(smesh);
+
+    //! define the time mesh
+    constexpr double dt_adv = 120.0; //!< Time step of advection problem
+    constexpr size_t NT = ReferenceScale::T / dt_adv + 1.e-4; //!< Number of Advections steps
+
+    //! MEVP parameters
+    constexpr double alpha = 1500.0;
+    constexpr double beta = 1500.0;
+    constexpr size_t NT_evp = 100;
+
+    //! Rheology-Parameters
+    Nextsim::VPParameters VP;
+
+    std::cout << "Time step size (advection) " << dt_adv << "\t" << NT << " time steps" << std::endl
+              << "MEVP subcycling NTevp " << NT_evp << "\t alpha/beta " << alpha << " / " << beta << std::endl
+              << "CG/DG/DGstress " << CG << "\t" << DGadvection << "\t" << DGstress
+              << std::endl;
+
+    //! VTK output
+    constexpr double T_vtk = 48. * 60.0 * 60.0; // only at end
+    constexpr size_t NT_vtk = T_vtk / dt_adv + 1.e-4;
+    //! LOG message
+    constexpr double T_log = 10.0 * 60.0; // every 30 minute
+    constexpr size_t NT_log = T_log / dt_adv + 1.e-4;
+
+    // netCDF output
+    std::string diagFile = "mevp.nc";
+    size_t ncOutputPeriod = 30; // Every hour
+
+    ////////////////////////////////////////////////// Forcing
+    Nextsim::Interpolations::Function2CG(smesh, momentum.GetOceanx(), OceanX());
```

Comment:
Aside from hiceName, these is all taken directly from the copy/paste source. I'm disinclined to make arbitrary changes solely in this file.
And it might help to think of hice as a word in and of itself. Like the plural of house!